lapce-app/src/doc.rs

```diff
@@ -696,8 +718,25 @@ impl Doc {
                     if current_rev != rev || doc.is_pristine() {
                         return;
                     }
-
-                    doc.save(|| {});
+                    if format {
+                        let send = create_ext_action(scope, move |result| {
+                            if let Ok(ProxyResponse::GetDocumentFormatting {
+                                edits,
+                            }) = result
+                            {
+                                let current_rev = doc.rev();
+                                if current_rev == rev {
+                                    doc.do_text_edit(&edits);
+                                }
+                            }
+                            doc.save(|| {});
```

Comment:
You need to check doc rev and return early if not matching. You can't run save if the doc has changed.